### PR TITLE
feat: detect the theme storage key of vitepress

### LINKED ISSUE #202: [bug] appearance/theme can't consist with vitepress
## descriptions

appearance/theme can't consist with vitepress

https://github.com/vuejs/devtools-next/assets/50254496/9f97bf4b-927c-48ed-bd65-2130a8a93a30

## repro

URL: <https://stackblitz.com/edit/vite-yfew9s?file=docs%2F.vitepress%2Fconfig.ts>

## additional context

It works with vitesse after updating deps.

### PR DESCRIPTION
**Pull request by LoTwT**
## description

This PR is to close #202.

It detects the theme storage key `vitepress-theme-appearance`. If the key is used in the local storage, it will be passed to `useColorMode`.

## question

Should the default key `vueuse-color-scheme` be synchronized with the `vitepress-theme-appearance` in a vitepress project?

### CODE REVIEW

**Comment by autofix-troubleshooter** [comment]
Hi! I'm the <a href="https://autofix.ci">autofix logo<strong>autofix.ci</strong></a> troubleshooter bot.

It looks like you correctly set up a CI job that uses the autofix.ci GitHub Action, but the [**autofix.ci GitHub App**](https://github.com/apps/autofix-ci/) has not been installed for this repository. This means that autofix.ci unfortunately does not have the permissions to fix this pull request. If you are the repository owner, please install the app and then restart the CI workflow! 😃

File: packages/ui/src/composables/theme.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  vitepress: 'vitepress-theme-appearance',
+  default: 'vueuse-color-scheme',
+}
+
+export function useTheme<T extends string = BasicColorMode>(options?: UseColorModeOptions<T>) {
+  options = options || {}
+
+  if (!options?.storageKey) {
+    const keys = Object.values(ThemeStroageKeys)
+    options.storageKey = keys.find(k => !!useLocalStorage(k, undefined).value)
```
Comment (alexzhang1030) [inline packages/ui/src/composables/theme.ts:16]: `useLocalStorage` is unnecessary here, just use `localStorage.getItem`

File: packages/ui/src/composables/theme.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * priorites following the desc order
+ */
+export const ThemeStroageKeys = {
+  vitepress: 'vitepress-theme-appearance',
+  default: 'vueuse-color-scheme',
+}
+
+export function useTheme<T extends string = BasicColorMode>(options?: UseColorModeOptions<T>) {
```
Comment (alexzhang1030) [inline packages/ui/src/composables/theme.ts:11]: I guess you should find a way to detect the environment. These two keys might both exist.

**Comment by alexzhang1030** [comment]
You should rebase main branch to fix the conflicts

File: packages/ui/src/composables/theme.ts (hunk 2 above)
Comment (LoTwT) [reply packages/ui/src/composables/theme.ts:11]: `Array.prototype.find()` follows the asc order to find the target item. So with the `Object.values(ThemeStorageKeys)`, the priority of `vitepress-theme-appearance` is higher than `vueuse-color-scheme`.
However, finding a better way to detect the environment is hard. I will try.

**Review by alexzhang1030 — APPROVED** [review]

**Review by songjianet — APPROVED** [review]

**Comment by webfansplz** [comment]
Thanks for the PR, but I think it's too special and out of scope for the devtools.

**Pull request closed without merging**